src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,236 @@
+
+PCRPrimerDesignForDNAAssemblyOPWidget::PCRPrimerDesignForDNAAssemblyOPWidget(AnnotatedDNAView* _annDnaView)
+    : QWidget(nullptr),
+      annDnaView(_annDnaView),
+      savableWidget(this, GObjectViewUtils::findViewByName(annDnaView->getName())) {
+    setupUi(this);
+    minMaxValuesSb = { { sbMinRequireGibbs, sbMaxRequireGibbs },
+                          { spMinRequireMeltingTeml, spMaxRequireMeltingTeml },
+                          { spMinRequireOverlapLength, spMaxRequireOverlapLength },
+                          { sbMinExcludeGibbs, sbMaxExcludeGibbs },
+                          { spMinExcludeMeltingTeml, spMaxExcludeMeltingTeml },
+                          { spMinExcludeOverlapLength, spMaxExcludeOverlapLength },
+                          { sbLeftAreaStart, sbLeftAreaEnd },
+                          { sbRightAreaStart, sbRightAreaEnd } };
+
+    userPrimersShowHideGroup->init(USER_PRIMERS_SHOW_HIDE_ID, tr("User primers"), wgtUserPrimers, true);
+    parametersOfPrimingSequencesShowHideGroup->init(PARAMETERS_OF_PRIMING_SEQUENCES_SHOW_HIDE_ID,
+                                                    tr("Parameters of priming sequences"),
+                                                    wgtParametersOfPrimingSequences,
+                                                    true);
+    parameters2ExcludeInWholePrimersShowHideGroup->init(PARAMETERS_2_EXCLUDE_IN_WHOLE_PRIMERS_SHOW_HIDE_ID,
+                                                        tr("Parameters to exclude in whole primers"),
+                                                        wgtParameters2ExcludeInWholePrimers,
+                                                        true);
+    areasForPrimingShowHideGroup->init(SELECT_AREAS_FOR_PRIMING_SHOW_HIDE_ID,
+                                       tr("Select areas for priming search"),
+                                       wgtAreasForPriming,
+                                       true);
+    openBackboneSequenceShowHideGroup->init(OPEN_BACKBONE_SEQUENCE_SHOW_HIDE_ID,
+                                            tr("Open the backbone sequence"),
+                                            wgtOpenBackboneSequence,
+                                            true);
+    generateSequenceShowHideGroup->init(GENERATE_SEQUENCE_SHOW_HIDE_ID,
+                                        tr("Generate a sequence"),
+                                        wgtGenerateSequence,
+                                        true);
+    otherSequencesInPCRShowHideGroup->init(OTHER_SEQUENCES_IN_PCR_REACTION_SHOW_HIDE_ID,
+                                           tr("Other sequences in PCR reaction"),
+                                           wgtOtherSequencesInPcr,
+                                           true);
+
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
```

Comment:
nit: This check is not needed. annDnaView can't be opened without at least 1 sequence.

Reply:
I this case, `SAFE_POINT` - the best way to check that this is true

Reply:
From the documentation of the AnnotatedDNAView I see:
"AnnotatedDNAView always has at least some 'active' sequence widget unless it is in the destroying phase."

It means that you check something unneeded.

Reply:
The check's been removed